src/unix/thread.c

```diff
@@ -646,13 +648,16 @@ void uv_cond_wait(uv_cond_t* cond, uv_mutex_t* mutex) {
 int uv_cond_timedwait(uv_cond_t* cond, uv_mutex_t* mutex, uint64_t timeout) {
   int r;
   struct timespec ts;
+  struct timeval tv;
 
 #if defined(__APPLE__) && defined(__MACH__)
   ts.tv_sec = timeout / NANOSEC;
   ts.tv_nsec = timeout % NANOSEC;
   r = pthread_cond_timedwait_relative_np(cond, mutex, &ts);
 #else
-  timeout += uv__hrtime(UV_CLOCK_PRECISE);
+  if (gettimeofday(&tv, NULL))
+    abort();
+  timeout += (tv.tv_sec * NANOSEC) + (tv.tv_usec * MICROSEC);
```

Comment:
Dimensional analysis:
timeout: This starts in nanoseconds.
timeout += (seconds in nanoseconds) + (microseconds * microseconds/second)

I think the intent is to convert tv.tv_usec to nanoseconds, in which case it should multiply by NANOSEC/MICROSEC (i.e. 1e3).

Reply:
Ooops! will change that :-)